invenio_app_rdm/utils/files.py

```diff
@@ -0,0 +1,145 @@
+    UI_URL = current_app.config.get("SITE_UI_URL")
+    return link_for(UI_URL, tpl, **kwargs)
+
+
+def link_for(base, tpl, **kwargs):
+    """Create a link using specific template."""
+    tpl = URLS.get(tpl)
+    for k in [
+        "key",
+    ]:
+        if k in kwargs:
+            kwargs[k] = quote(kwargs[k].encode("utf8"))
+    return tpl.format(base=base, **kwargs)
+
+
+def format_file_integrity_report(report):
+    """Format the email body for the file integrity report."""
+    lines = []
+    for entry in report:
+        f = entry["file"]
+        lines.append("ID: {}".format(str(f.id)))
+        lines.append("URI: {}".format(f.uri))
+        lines.append("Name: {}".format(entry.get("filename")))
+        lines.append("Created: {}".format(f.created))
+        lines.append("Checksum: {}".format(f.checksum))
+        lines.append("Last Check: {}".format(f.last_check_at))
+        if "record" in entry:
+            lines.append(
+                "Record: {}".format(
+                    ui_link_for("record_html", id=entry["record"].get("id"))
+                )
+            )
+        if "deposit" in entry:
+            lines.append(
+                "Deposit: {}".format(
+                    ui_link_for("deposit_html", id=entry["deposit"].get("id"))
+                )
+            )
+        lines.append(("-" * 80) + "\n")
+    return "\n".join(lines)
+
+
+def generate_file_integrity_report(files):
+    """Generates a file integrity report."""
+    report = []
+    for file in files:
+        entry = _generate_report_entry(file)
+        report.append(entry)
+    return report
+
+
+def _generate_report_entry(file):
+    """Generates a single entry for the files integrity report."""
+    entry = {"file": file}
+    for o in file.objects:
+        entry["filename"] = o.key
+
+        # Find records/deposits for the files
+        deposit = get_deposit_from_bucket(o.bucket_id)
+        if deposit and deposit.json:
+            entry["deposit"] = deposit.json
+        else:
+            record = get_record_from_bucket(o.bucket_id)
+            if record and record.json:
+                entry["record"] = record.json
+    return entry
+
+
+def get_record_from_bucket(bucket_id):
+    """Retrieve a record from a bucket id."""
+    return RDMRecordMetadata.query.filter_by(bucket_id=bucket_id).one_or_none()
+
+
+def get_deposit_from_bucket(bucket_id):
+    """Retrieve a deposit from a bucket id."""
+    return RDMDraftMetadata.query.filter_by(bucket_id=bucket_id).one_or_none()
+
+
+def send_integrity_report_email(report):
+    """Sends an e-mail with the given report.
+
+    APP configs ('SECURITY_EMAIL_SENDER', 'ADMIN_EMAIL') must be set.
+    Oherwise a warning is logged.
+    """
+    try:
+        if report:
+            subject = "Zenodo files integrity report [{}]".format(datetime.now())
+            body = format_file_integrity_report(report)
+            sender = current_app.config.get("SECURITY_EMAIL_SENDER")
```

Comment:
Both configs `SECURITY_EMAIL_SENDER` and `ADMIN_EMAIL` must be set in order to receive an e-mail.

Let me know if there are more appropriate configs. Otherwise, this should be documenented.

Reply:
minor: since we're using `Flask-Mail` under the hood, it might be a good idea to set `MAIL_DEFAULT_SENDER` (which would also get picked up by Flask-Security). I'm not sure why we're not setting this globally somewhere instead of the more specific `SECURITY_EMAIL_SENDER`. If that was the case we wouldn't need to pass `sender` at all.


For `ADMIN_EMAIL`,

Reply:
minor: since we're using `Flask-Mail` under the hood, it might be a good idea to set `MAIL_DEFAULT_SENDER` (which would also get picked up by Flask-Security). I'm not sure why we're not setting this globally somewhere instead of the more specific `SECURITY_EMAIL_SENDER`. If that was the case we wouldn't need to pass `sender` at all.

For `ADMIN_EMAIL`,

Reply:
About `MAIL_DEFAULT_SENDER` I am a bit sceptical to change it in this PR as it would possibly affect others. But it is a valid point that `SECURITY_EMAIL_SENDER`, if not set, defaults to `MAIL_DEFAULT_SENDER` because we're using `Flask-Mail` anyway.

Reply:
I would indeed use `MAIL_DEFAULT_SENDER`, but make sure that both `MAIL_DEFAULT_SENDER` and `SECURITY_EMAIL_SENDER` are documented in the docs.